tests/test_builders/test_build_linkcheck.py

```diff
@@ -216,12 +244,20 @@ def test_too_many_retries(app):
     assert row['lineno'] == 1
     assert row['status'] == 'broken'
     assert row['code'] == 0
-    assert row['uri'] == 'https://localhost:7777/doesnotexist'
+    assert row['uri'] == f'https://{netloc}/doesnotexist'
 
 
 @pytest.mark.sphinx('linkcheck', testroot='linkcheck-raw-node', freshenv=True)
 def test_raw_node(app):
-    with http_server(OKHandler):
+    with http_server(OKHandler) as port, rewrite_netlocs(app, port) as netloc:
+        # write an index file that contains a link back to this webserver's root
+        # URL.  docutils will replace the raw node with the contents retrieved..
+        # ..and then the linkchecker will check that the root URL is available.
+        index = (app.srcdir / "index.rst")
+        index.write_text(
+            ".. raw:: 'html'\n"
+            "   :url: http://{}/\n".format(netloc),
```

Comment:
```suggestion
            "   :url: http://{netloc}/".format(netloc=netloc),
```

(reduced symbol soup)